File: EAGetMail/Samples_NetDesktop/VB/ImapFull/AppProject1/FormSelectFolder.vb
Imports EAGetMail

Public Class FormSelectFolder
    Public OriginalFolderPath As String = ""

    Private Sub ButtonOK_Click(sender As Object, e As EventArgs) Handles ButtonOK.Click
        Dim node As TreeNode = TreeViewFolder.SelectedNode

        If node Is Nothing OrElse node.Parent Is Nothing Then
            MessageBox.Show("please select a folder.")
            Me.DialogResult = DialogResult.None
            Return
        End If

        Dim folder As Imap4Folder = TryCast(node.Tag, Imap4Folder)

        If String.Compare(folder.ServerPath, OriginalFolderPath, True) = 0 Then
            MessageBox.Show("please select a folder different with original folder.")
            Me.DialogResult = DialogResult.None
            Return
        End If

        Me.DialogResult = DialogResult.OK
    End Sub
End Class

File: EAGetMail/Samples_NetDesktop/CSharp/ImapFull/AppProject1/FormSelectFolder.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using EAGetMail;

namespace AppProject1
{
    public partial class FormSelectFolder : Form
    {
        public string OriginalFolderPath = "";
        public FormSelectFolder()
        {
            InitializeComponent();
        }

        private void ButtonOK_Click(object sender, EventArgs e)
        {
            TreeNode node = TreeViewFolder.SelectedNode;
            if (node == null || node.Parent == null)
            {
                MessageBox.Show("please select a folder.");
                this.DialogResult = DialogResult.None;
                return;
            }

            Imap4Folder folder = node.Tag as Imap4Folder;
            if (string.Compare(folder.ServerPath, OriginalFolderPath, true) == 0)
            {
                MessageBox.Show("please select a folder different with original folder.");
                this.DialogResult = DialogResult.None;
                return;
            }

            this.DialogResult = DialogResult.OK;
        }
    }
}
